### ISSUE TITLE
For some reason, issues are created twice?

### ISSUE THREAD

**Issue by camposcristianeze** [alstr/todo-to-issue-action#10]
This is my current workflow.

Thanks!

```
# This is a basic workflow to help you get started with Actions

name: TODO's to issues

# Controls when the action will run. Triggers the workflow on push or pull request
# events but only for the master branch
on:
  pull_request:
    branches: [ master, staging ]

# A workflow run is made up of one or more jobs that can run sequentially or in parallel
jobs:
  build:
    runs-on: "ubuntu-latest"
    steps:
      - uses: "actions/checkout@master"
      - name: "TODO to Issue"
        uses: "alstr/todo-to-issue-action@master"
        with:
          REPO: ${{ github.repository }}
          BEFORE: ${{ github.event.before }}
          SHA: ${{ github.sha }}
          TOKEN: ${{ secrets.GITHUB_TOKEN }}
          LABEL: "@todo"
          COMMENT_MARKER: "@"
        id: "todo"
```

**Comment by alstr** [comment]
Hey Cristian. 👋

Thanks for the report. I've just done a quick test and I can't reproduce this. Would you be able to share logs?

Workflow events do potentially encompass several activities, which might lead to them running more than once. This page outlines the activities that trigger each event: https://help.github.com/en/actions/reference/events-that-trigger-workflows#pull-request-event-pull_request

You may want to use the `types` keyword to narrow down the events, as described here: https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#onevent_nametypes

For example:

```
on:
  release:
    # Only use the types keyword to narrow down the activity types that will trigger your workflow.
    types: [published, created, edited]
```

Also, I notice your workflow file says `Triggers the workflow on push or pull request`, but you've only set it to run on `pull_request` - is that intentional?

**Comment by camposcristianeze** [comment]
Great, thanks for the context! Attaching some logs, but not sure if they're very relevant. Let me know if you need something specific. 

[logs_2363.zip](https://github.com/alstr/todo-to-issue-action/files/4675486/logs_2363.zip)
[logs_2358.zip](https://github.com/alstr/todo-to-issue-action/files/4675487/logs_2358.zip)

Also, I thought if we run the action on every push on staging/master, issues won't be created again if they already exist? I could try using on: release for every time we create a new one, but if we haven't tackled the TODO and deleted the comment it will be fired again?

> Also, I notice your workflow file says Triggers the workflow on push or pull request, but you've only set it to run on pull_request - is that intentional?

Sorry, this was just a comment left behind :)

**Comment by alstr** [comment]
Hi Cristian, thanks for the feedback.

From looking at the logs you provided, I can't immediately see anything wrong. The action ran once, successfully, in both cases.

You are correct that if the TODO has already been added _in an earlier commit,_ it shouldn't be added again. However, if, for some reason, the same commit was processed by an action multiple times, it could theoretically create multiple versions of the same TODO. I haven't encountered that yet, as I have really only used it with `on: push`.

My suggestion would be, if you are able, try using `on: push`. Then, if that works and you want to refine how the action runs, gradually alter the workflow file. When the duplication issue occurs, let me know what changes caused it and I can take another look. A code snippet might be helpful also.

Like I said, at the moment the action is designed to work around `on: push`. Anything else is uncharted territory, but I would be willing to look into it or open to PRs!

Thanks!

**Comment by cynthiarich** [comment]
@alstr I saw similar behavior today. Here's some links to check it out:

- My workflow file is [here](https://github.com/coding-with-cynthia/coding-with-cynthia.github.io/blob/master/.github/workflows/get-todos.yml) running only on `[push]`
- I added a TODO in the same PR [here](https://github.com/coding-with-cynthia/coding-with-cynthia.github.io/pull/10/commits/4a1fd1d88fb741073b376539a03fa92dccf93024) 
- It created the issue [here](https://github.com/coding-with-cynthia/coding-with-cynthia.github.io/issues/11) and another one [here](https://github.com/coding-with-cynthia/coding-with-cynthia.github.io/issues/12)

It's possible it is picking it up the merge to master as a second commit, but I thought it was supposed to ignore it since there was already an issue opened for that comment.

It also didn't close the issue when I removed the comment 😢

**Comment by alstr** [comment]
Hi @cynthiarich. 👋

I have released a new version that may resolve the issue duplication you and @camposcristianeze reported.

I think what you suggested is probably correct. Previously, the action only checked against the commit diff file to see if a TODO already existed, as the action was built pretty quickly and hadn't been tested against merges. In some circumstances that could perhaps lead to duplicates. I've added a hash based on the issue body to the end of all new issues created by the action. When processing TODOs in future, it will check if an issue already exists with this hash.

If you make sure your `.workflow` file points to `alstr/todo-to-issue-action@master`, you should be able to try it out. Let me know if it helps.

Regarding the failure to close the issue, this is still quite an experimental feature, and I'm not 100% sure why it didn't work in your case. I have made a few amends as part of the above, so maybe try it again and see. If it still doesn't work as intended, please create a new issue with a link to the action log and commit in question. Thanks. 😄

**Comment by alstr** [comment]
Closing this due to lack of activity but if there's still an issue do let me know. :smile:
